corehq/apps/users/models.py

```diff
@@ -1366,6 +1366,16 @@ def set_location(self, location):
 
         self.save()
 
+    def unset_location(self):
+        """
+        Unset the location and remove all associated user data and cases
+        """
+        self.user_data.pop('commtrack-supply-point', None)
+        self.user_data.pop('commcare_primary_case_sharing_id', None)
+        self.location_id = None
+        self.clear_locations()
+        self.save()
```

Comment:
@czue wrote:
> I think you might need to add a check that is something like:
>```
if self.project.supports_multiple_locations_per_user:
    self.clear_locations()
```
> though it also looks like there might be some problems with that function
I think I actually need to call `clear_locations` regardless of multiple locations - even with a single location, there is a location_map_case being created (perhaps needlessly?).